src/small_strucures.c

```diff
@@ -502,3 +503,31 @@ void sub_08063E64(struct Unk_03006AF4 *r0)
     sub_08011104(9);
     sub_08011104(10);
 }
+
+extern void (*gUnk_08BC9C68[])(s16 a, s16 b, s16 c, s16 d);
+extern s16 (*gUnk_08BC9C7C[])(void);
+extern u16 sub_08068070(u16);
+
+s16 sub_08063F70(void)
+{
+    u16 r3,r4,r5,r6;
+    struct Unk_03006AF4 *ptr = gUnk_03006AF4;
+    switch ((s16)ptr->unk2) // TODO:figure out why this weird cast keeps to happen
```

Comment:
Maybe it's really s16?

Reply:
It's not matching with s16😳

Reply:
> It's not matching with s16😳

Where?

Reply:
https://cexplore.karathan.at/z/c2Za1u
I think it's the function array part.

Reply:
Then you can cast it to u16 instead. s16 -> u16 is implementation-defined, but s16 -> u16 is always well-defined.